beard/metrics/clustering.py

```diff
@@ -57,21 +57,39 @@ def paired_precision_recall_fscore(labels_true, labels_pred):
         raise ValueError(
             "input labels must not be empty.")
 
-    # Reconstruct the given input by grouping the samples
-    c_gold = _group_samples_by_cluster_id(labels_true)
-    c_system = _group_samples_by_cluster_id(labels_pred)
-
-    # Calculate pairs (list of tuples)
-    c_gold_pairs = _calculate_pairs(c_gold)
-    c_system_pairs = _calculate_pairs(c_system)
-
-    # Calculate evaluation metrics
-    inters = c_gold_pairs.intersection(c_system_pairs)
-
-    precision = len(inters)/len(c_system_pairs)
-    recall = len(inters)/len(c_gold_pairs)
-    # (precision+recall) always > 0
-    harmonic_mean = 2*precision*recall/(precision + recall)
+    f_one = lambda x, y: 1.0
+    f_zero = lambda x, y: 0.0
+    f_mult = lambda x, y: x * y
+
+    # Calculate precision
+    numerator = _general_merge_distance(labels_true, labels_pred,
+                                        fm=f_zero, fs=f_mult)
+    denominator = _general_merge_distance(range(len(labels_pred)),
+                                          labels_pred,
+                                          fm=f_zero, fs=f_mult)
```

Comment:
`xrange` is not Python 3 compatible. Plus, in this case, we will have anyway to generate all values and hold them into memory, so there is no point in using a generator instead.